src/mono/mono/utils/lifo-semaphore.h

```diff
@@ -0,0 +1,34 @@
+#ifndef __MONO_LIFO_SEMAPHORE_H__
+#define __MONO_LIFO_SEMAPHORE_H__
+
+#include <mono/utils/mono-coop-mutex.h>
+
+typedef struct _LifoSemaphore LifoSemaphore;
```

Comment:
Can we maybe drop the leading underscore:
Pro underscore: It means don't use that identifier, it is private, i.e. say `foo`, not `struct _foo`.
Con underscore: Consider what the C++ would look like.